# <center>Exploratory Data Analysis of the Disney datasets

## Foreward

This notebook will be showing some exploratory data analysis for the `DISNEY` dataset located at https://data.world/kgarrett/disney-character-success-00-16. 

# Introduction

## Question(s) of interests
In this analysis, I will investigate a question associated with Disney movies. I am interested in finding out which genre of Disney movies is the most popular. To answer this question, I will determine the genre in which the maximum number of movies have been produced and which genre has generated the highest gross income for Disney. My next question is to find out which of the five subsidiaries of Disney generates the maximum revenue and which of these seems to be a promising source of revenue in the future.

## Dataset description 

The below descripitions were taken directly from the [website](https://en.wikipedia.org/wiki/The_Walt_Disney_Company#2005%E2%80%932020:_Bob_Iger's_leadership,_expansion,_and_Disney+ ).

"The Walt Disney Company, commonly known as Disney is an American multinational, mass media and entertainment conglomerate that is headquartered at the Walt Disney Studios complex in Burbank, California. Disney was founded on October 16, 1923, by brothers Walt and Roy O. Disney is one of the world's largest entertainment companies and is considered to be a pioneer in the animation industry, having produced 790 features, 122 of which are animated films, and has been ranked number 53 on the 2022 Fortune 500 list of biggest companies in the United States by revenue." 

The Disney dataset is composed of $5$ tables, `disney_movies_total_gross.csv`, `disney_revenue_1991-2016.csv`, `disney-characters.csv`, `disney-director.csv` and `disney-voice-actors.csv`. Each table is stored in a `.csv` file and contains different information about Disney movies including characters, directors, voice actors and revenue. I will be using the `disney_movies_total_gross` and `disney_revenue_1991-2016` tables formally described below: 

* **disney_movies_total_gross.csv** 
    * This file includes information on Disney gross. It tells the total gross income and the inflation adjusted gross for each of the movie, along with its release date, and the genre of the movies with its MPAA rating. 
* **disney_revenue_1991-2016.csv**
    * This file includes information on the revenue generated by five subsidiary companies of Disney. It tells the revenue generated by Studio Entertainment, Disney Consumer Products, Disney Interactive, Walt Disney Parks and Resorts and Disney Media Networks from the year 1991 to 2016. 

# Methods and Results

As I am interested in the information regarding the gross income and revenue, I will be using tables **disney_movies_total_gross** and **disney_revenue_1991-2016**.

I will start by importing the tables and doing some basic cleaning of data to make the tables ready for our visualizations.

In [1]:
# Lets import all the required libraries needed for this analysis
import altair as alt
import pandas as pd

# import all the required files
movies = pd.read_csv("data/disney_movies_total_gross.csv")
revenue = pd.read_csv("data/disney_revenue_1991-2016.csv")

Let's see how the tables look like.

In [2]:
#displaying the first five rows of the dataframe
movies.head()

movie_title  release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                        Pinocchio   Feb 9, 1940  Adventure           G   
2                         Fantasia  Nov 13, 1940    Musical           G   
3                Song of the South  Nov 12, 1946  Adventure           G   
4                       Cinderella  Feb 15, 1950      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

In [3]:
#displaying the first five rows of the dataframe
revenue.head()

Year  Studio Entertainment[NI 1]  Disney Consumer Products[NI 2]  \
0  1991                      2593.0                           724.0   
1  1992                      3115.0                          1081.0   
2  1993                      3673.4                          1415.1   
3  1994                      4793.0                          1798.2   
4  1995                      6001.5                          2150.0   

   Disney Interactive[NI 3][Rev 1]  Walt Disney Parks and Resorts  \
0                              NaN                         2794.0   
1                              NaN                         3306.0   
2                              NaN                         3440.7   
3                              NaN                         3463.6   
4                              NaN                         3959.8   

  Disney Media Networks  Total  
0                   NaN   6111  
1                   NaN   7502  
2                   NaN   8529  
3                   359  10414  
4                   414  12525

Let's get some more detailed information about the **movies** table.

In [4]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   movie_title               579 non-null    object
 1   release_date              579 non-null    object
 2   genre                     562 non-null    object
 3   MPAA_rating               523 non-null    object
 4   total_gross               579 non-null    object
 5   inflation_adjusted_gross  579 non-null    object
dtypes: object(6)
memory usage: 27.3+ KB


The movies table has $579$ rows and $6$ columns. Every **movie_title** has the **release_date**, the **genre**, the **MPAA_rating**, the **total_gross** and also the **inflation_adjusted_gross**

The **total_gross** and **inflation_adjusted_gross** columns in the movies table have numeric values, however, they are saved as 'object'. We will clean the above columns to get a better visualization of the data.

In [5]:
#import the function from the python script
from my_functions import clean_str

#clean the columns
movies_cleaned = clean_str(movies,"total_gross")
movies_cleaned = clean_str(movies,"inflation_adjusted_gross")
movies_cleaned.dtypes

movie_title                  object
release_date                 object
genre                        object
MPAA_rating                  object
total_gross                 float64
inflation_adjusted_gross    float64
dtype: object

Also, the values in **total_gross** and **inflation_adjusted_gross** are very large. So, we will the reorganize the above table into US million dollars, rounded to two decimal places.

In [6]:
#change the gross to US million dollars and round to 2 decimal places
movies_cleaned["total_gross"]=movies_cleaned["total_gross"].div(1000000).round(2)
movies_cleaned["inflation_adjusted_gross"]=movies_cleaned["inflation_adjusted_gross"].div(1000000).round(2)

In [7]:
movies_cleaned.head(2)

movie_title  release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                        Pinocchio   Feb 9, 1940  Adventure           G   

   total_gross  inflation_adjusted_gross  
0       184.93                   5228.95  
1        84.30                   2188.23

Now, we have a clean dataframe for **movies**.

We will generate some information about the **revenue** table.

In [8]:
revenue.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26 entries, 0 to 25
Data columns (total 7 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Year                             26 non-null     int64  
 1   Studio Entertainment[NI 1]       25 non-null     float64
 2   Disney Consumer Products[NI 2]   24 non-null     float64
 3   Disney Interactive[NI 3][Rev 1]  12 non-null     float64
 4   Walt Disney Parks and Resorts    26 non-null     float64
 5   Disney Media Networks            23 non-null     object 
 6   Total                            26 non-null     int64  
dtypes: float64(4), int64(2), object(1)
memory usage: 1.5+ KB


The **revenue** table has $26$ rows with $7$ columns. For each **year**, we are given the revenue from **Studio Entertainment[NI 1]**, **Disney Consumer Products[NI 2]**, **Disney Interactive[NI 3]**, **Walt Disney Parks and Resorts** and **Disney Media Networks**. This gives the total revenue generated each year by Disney.

The revenue table has one of the columns **Disney Media Networks** as an object which would not allow us to use the numeric values from the column. Let us change it to float first using the **clean_str** function which is a custom function in the script that takes in a dataframe and a specific column from the dataframe and removes all the non-numeric characters and changes the data type to float.

In [9]:
revenue_cleaned = clean_str(revenue,"Disney Media Networks")
revenue_cleaned.dtypes

Year                                 int64
Studio Entertainment[NI 1]         float64
Disney Consumer Products[NI 2]     float64
Disney Interactive[NI 3][Rev 1]    float64
Walt Disney Parks and Resorts      float64
Disney Media Networks              float64
Total                                int64
dtype: object

The column names of the above data have square brackets in them, we will restructure the columns to remove those square brackets.

In [10]:
#changing the name of columns to remove the square brackets
revenue_cleaned = revenue_cleaned.rename(columns = {"Studio Entertainment[NI 1]":"Studio Entertainment",
                                           "Disney Consumer Products[NI 2]":"Disney Consumer Products",
                                           "Disney Interactive[NI 3][Rev 1]":"Disney Interactive"})
revenue_cleaned.head(2)

Year  Studio Entertainment  Disney Consumer Products  Disney Interactive  \
0  1991                2593.0                     724.0                 NaN   
1  1992                3115.0                    1081.0                 NaN   

   Walt Disney Parks and Resorts  Disney Media Networks  Total  
0                         2794.0                    NaN   6111  
1                         3306.0                    NaN   7502

The last step to have a clean dataframe, we will replace the **null** values in the columns of the dataframe.

In [11]:
#import the function to clean the dataframe to fill in the null values
from my_functions import fill_null_values
revenue_cleaned = fill_null_values(revenue_cleaned)
revenue_cleaned.head(2)

Year  Studio Entertainment  Disney Consumer Products  Disney Interactive  \
0  1991                2593.0                     724.0               174.0   
1  1992                3115.0                    1081.0               174.0   

   Walt Disney Parks and Resorts  Disney Media Networks  Total  
0                         2794.0                  359.0   6111  
1                         3306.0                  359.0   7502

Now that we have a clean dataframe, we can start with the visualizations.

As a first visualization, let us look at the number of movies released for each genre. To obtain this information, I will use the **movies** table. I will group them by genre and then find the size of each group to see how many movies of each genre have been released by Disney.

In [12]:
# group by genre and count the number of movies for each group
genre_group=movies_cleaned.groupby('genre').count().loc[:,"movie_title"]

# Reset the index so we can plot using altair
genre_group = genre_group.reset_index()

#Renaming the column to number of movies
genre_group = genre_group.rename(columns = {"movie_title": "number of movies"})
genre_group

genre  number of movies
0                Action                40
1             Adventure               129
2          Black Comedy                 3
3                Comedy               182
4   Concert/Performance                 2
5           Documentary                16
6                 Drama               114
7                Horror                 6
8               Musical                16
9       Romantic Comedy                23
10    Thriller/Suspense                24
11              Western                 7

As we have the the number of movies in each genre, a bar plot can be created to visualize it.

In [13]:
# Use altair to generate a bar plot
genre_plot = (
    alt.Chart(genre_group, width=500, height=300)
    .mark_bar()
    .encode(
        x=alt.X("genre:O", sort="y",title="Movie Genre"),
        y=alt.Y("number of movies:Q", title="Number of movies"),
    )
    .properties(title="Number of Movies released in each genre")
)
genre_plot

alt.Chart(...)

From the above plot, maximum number of movies have been released in the **Comedy** genre, followed by **Adventure** and then **Drama**. 

Since Disney has produced maximum number of movies in the **Comedy** genre, let us check whether it creates maximum income for the company.

In [14]:
# Group the dataframe by genre and sum the total_gross column
genre_total_gross = movies_cleaned.groupby('genre')['total_gross'].sum()

# Create a new dataframe with the total gross for each genre
genre_df = pd.DataFrame(genre_total_gross).reset_index()

# Print the new dataframe
genre_df

genre  total_gross
0                Action      4184.56
1             Adventure     16389.02
2          Black Comedy        97.54
3                Comedy      8119.62
4   Concert/Performance       103.45
5           Documentary       180.69
6                 Drama      4106.95
7                Horror        87.08
8               Musical      1157.29
9       Romantic Comedy      1152.23
10    Thriller/Suspense      1406.81
11              Western       359.02

Now that we have our values, let's generate a bar plot.

In [15]:
gross_plot = (
        alt.Chart(genre_df, width=500, height=300)
        .mark_bar()
        .encode(
            x=alt.X("genre:O",sort="-y", title="Genre"),
            y=alt.Y("total_gross:Q", title="Gross in U.S Million Dollars"),
         )
        .properties(title="Gross income through different genres of movies")
    )

gross_plot

alt.Chart(...)

To our surprise even though the maximum number of movies have been produced in the **Comedy** genre by Disney, the **Adventure** genre has generated the maximum gross. 

I would also like to see which genre has the maximum gross `adjusted to inflation` as of 2016.

In [16]:
# Group the dataframe by genre and sum the inflation_adjusted_gross column
genre_inflated_gross = movies_cleaned.groupby('genre')['inflation_adjusted_gross'].sum()

# Create a new dataframe with the total gross for each genre
genre_df2 = pd.DataFrame(genre_inflated_gross).reset_index()

# Print the new dataframe
genre_df2

genre  inflation_adjusted_gross
0                Action                   5498.94
1             Adventure                  24561.23
2          Black Comedy                    156.73
3                Comedy                  15409.52
4   Concert/Performance                    114.82
5           Documentary                    203.49
6                 Drama                   8195.78
7                Horror                    140.49
8               Musical                   9657.57
9       Romantic Comedy                   1788.88
10    Thriller/Suspense                   2151.66
11              Western                    516.71

Let's plot the graph to visualize it.

In [17]:
gross_plot = (
        alt.Chart(genre_df2, width=500, height=300)
        .mark_bar()
        .encode(
            x=alt.X("genre:O",sort="-y", title="Genre"),
            y=alt.Y("inflation_adjusted_gross:Q", title="Inflation adjusted gross in U.S Million Dollars"),
         )
        .properties(title="Inflation adjusted gross through different genres of movies")
    )

gross_plot

alt.Chart(...)

Clearly, the highest gross earner for Disney (both **total_gross** and **inflation_adjusted_gross**) has been the **Adventure** genre.

Just out of curiosity, we will find out if the movie which has generated the maximum income is from `Adventure` genre or any other genre.

In [18]:
max_gross = movies_cleaned["total_gross"].max()
max_gross_rows = movies_cleaned[movies_cleaned["total_gross"] == max_gross]
max_gross_rows

movie_title  release_date      genre  \
564  Star Wars Ep. VII: The Force Awakens  Dec 18, 2015  Adventure   

    MPAA_rating  total_gross  inflation_adjusted_gross  
564       PG-13       936.66                    936.66

Let's see if the **inflation adjusted gross** affects this output.

In [19]:
max_gross = movies_cleaned["inflation_adjusted_gross"].max()
max_gross_rows = movies_cleaned[movies_cleaned["inflation_adjusted_gross"] == max_gross]
max_gross_rows

movie_title  release_date    genre MPAA_rating  \
0  Snow White and the Seven Dwarfs  Dec 21, 1937  Musical           G   

   total_gross  inflation_adjusted_gross  
0       184.93                   5228.95

The movie which has generated the highest gross income for Disney is **Star Wars Ep VII: The Force Awakens**, released in 2015, which is an **Adventure** movie. However, if we look at the inflation-adjusted gross, then the movie with the maximum income is **Snow White and the Seven Dwarfs**, released in 1937, and from the genre **Musical**. Surprisingly, none of these is from the **Comedy** genre, which was the most popular genre, based on the numbers of movies Disney has produced. 

Completing the first dataframe about the gross income and its visualization, let us further explore the **revenue** dataframe. 

The Walt Disney Companies have five major business segments which generate revenue for Disney: **Studio Entertainment** ,**Disney Consumer Products**,**Disney Interactive**,**Walt Disney Parks and Resorts**, and **Disney Media Networks**. We will be finding which of these generates the maximum revenue for Disney. 

In [20]:
#finding the mean of required columns
average_revenue = revenue_cleaned[["Studio Entertainment",
                           "Disney Consumer Products",
                           "Disney Interactive",
                          "Walt Disney Parks and Resorts", "Disney Media Networks"]].mean()

# Reset the index so we can plot using altair
average_revenue = average_revenue.round(2).reset_index()

#changing the name of the columns
average_revenue=average_revenue.rename(columns={"index":"Name", 0 : "Mean"})

average_revenue

Name      Mean
0           Studio Entertainment   6465.65
1       Disney Consumer Products   2638.69
2             Disney Interactive    585.64
3  Walt Disney Parks and Resorts   8512.62
4          Disney Media Networks  11433.23

Now, plotting a graph to see the mean value of these five subsidiaries.

In [21]:
# Use altair to generate a bar plot
revenue_plot = (
    alt.Chart(average_revenue, width=500, height=300)
    .mark_bar()
    .encode(
        x=alt.X("Name:O", sort="-y",title="Business Segment"),
        y=alt.Y("Mean:Q", title="Mean"),
    )
    .properties(title="Revenue generated by different segments in Million U.S Dollars")
)
revenue_plot

alt.Chart(...)

We can observe that **Disney Media Networks** has generated the maximum revenue for Disney in the years 1991-2016.

Let's see the trend in the revenue generation for all the five subsidiaries over the years.

In [22]:
#list of the name of columns if the subsidiaries
list_of_columns = ['Studio Entertainment','Disney Consumer Products',
                   'Disney Interactive', 'Walt Disney Parks and Resorts', 'Disney Media Networks']
colors = ['blue', 'red', 'green', 'orange', 'purple']

In [23]:
#generating the line graph and using transform_fold to encode the 5 groups as the parameter

revenue_chart = (
    alt.Chart(revenue_cleaned, width=500, height=300)
    .transform_fold(
        list_of_columns,
        as_=['Category', 'Revenue']
    )
    .mark_line()
    .encode(
        x=alt.X("Year:O", title="Year"),
        y=alt.Y("Revenue:Q", title="Revenue in U.S Million Dollars"),
        color=alt.Color("Category:N", scale=alt.Scale(domain=list_of_columns, range=colors)),
    )
    .properties(title="Revenue generated by Disney through different segments")
)

revenue_chart


alt.Chart(...)

From the above line graphs, we can notice that **Disney Media Networks** has been consistently generating large amount of revenues for the company. The line graph is almost linear and depicts that for each year increase, the revenue by Disney Media Networks increases much more rapidly than any other subsidiary.

# Discussions

In this project, I have worked with the Disney dataset and tried to compute information about their gross income and the revenue generated from the year 1991 to the year 2016. 

I was interested in knowing the most popular genre of movies produced by Disney. I expected the highest-grosser movie to be from the `musical` genre like `Beauty and the Beast` which also happens to be favourite Disney movie. However, based on the number of movies produced, `Comedy` was the most popular genre with 182 movies, followed by Adventure with 129 movies produced. And further looking at the genre which has generated the most income for Disney, about 16.4 billion US Dollars, is **Adventure** genre with **Star Wars Ep. VII: The Force Awakens** being the movie which has the highest total gross. Considering inflation as of 2016, the maximum inflation-adjusted gross (about 24.6 billion US Dollars) has still been earned through Adventure movies.

Another question which I was interested in looking at, was the revenue generated by different subsidiaries of Disney. And undoubtedly, **Disney Media Networks** generated maximum revenue for the company, followed by Walt Disney Park and Resorts. I expected the highest revenue to be generated by Disney Media Networks as it holds a prominent position in the media industry with an extensive array of media and entertainment resources covering cable networks, broadcasting, and digital platforms. 

This analysis provides an idea of which movie genres seem to be more profitable for the company and which types of movies should be produced in the future to generate income. Additionally, Disney may want to explore ways to increase revenue from its subsidiary, `Disney Interactive`, to ensure consistent growth of the company.

To continue this analysis, another question that could be explored is the number of movies produced for different MPAA ratings, as well as the average gross income generated by movies under each rating. We can also investigate if the release date of the movie plays any role in the success of the movie. This will be interesting to determine if the rating of the movies and the release date are some of the factors which affect the gross income.

# References

Some of the content in this notebook is not entirely original, as certain sections were obtained from online resources. It is important to note that I do not claim credit for these parts, as they were solely utilized for the purpose of providing information.

## Resources used
* [Data Source](https://data.world/kgarrett/disney-character-success-00-16)
    * The Disney database used in this work was created by **Kelly Garrett**.
* [Data Visualization](https://altair-viz.github.io/user_guide/transform/fold.html#user-guide-fold-transform)
    * Inspiration for generating the the line graph of different subsidiaries on the same graph.
* [Facts about Disney](https://en.wikipedia.org/wiki/The_Walt_Disney_Company)
    * General information about Disney was gathered from Wikepedia
* [Facts about Disney Media Networks](https://en.wikipedia.org/wiki/Disney_Media_Networks)
    * General information about Disney Media netwrok was gathered from Wikepedia
* [Inspiration](https://github.com/UBC-MDS/prog-python-ds-students/blob/main/release/project_sample/project_sample.ipynb)
    * The inspiration was taken from the sample project

![](img/Walt_Disney_Studios_Alameda_Entrance.jpg)

Image obtained from [here](https://en.wikipedia.org/wiki/The_Walt_Disney_Company#/media/File:Walt_Disney_Studios_Alameda_Entrance.jpg)